hyejin/app/src/main/java/com/joocoding/android/app/githubsearch/network/ClientFactory.kt

```diff
@@ -25,15 +32,17 @@ object ClientFactory {
             }
         }.build()
 
-    private fun createRetrofit(): Retrofit =
+    @Provides
+    fun createRetrofit(): Retrofit =
         Retrofit.Builder().apply {
             baseUrl(BASE_URL)
             addConverterFactory(GsonConverterFactory.create())
             client(okHttpClient)
         }.build()
 
-    fun <T> createService(service: Class<T>): T {
-        return retrofit.create(service)
+    @Provides
+    fun createService(): SearchService {
+        return retrofit.create(SearchService::class.java)
```

Comment:
```suggestion
    fun createSearchService(): SearchService {
        return retrofit.create(SearchService::class.java)
```

추후 Service 객체가 많이 생성될 수 있으니까 네이밍 조금 수정되면 좋을 것 같네요!